### PR TITLE
ARROW-8609: [C++] Fix ORC Java JNI crash

### PR DESCRIPTION
**Pull request by zhouyuan**
check if arrow buffer is null before passing to the constructor

Signed-off-by: Yuan Zhou <yuan.zhou@intel.com>

### CODE REVIEW

**Comment by kszucs** [comment]
Thanks for working on this. Could you please [enable the ORC tests](https://github.com/apache/arrow/pull/6953/files#diff-3cfa30ba4048e90c3da87ba3eee27816R36) in the jni build (just use `java/adapter/orc`)?

**Comment by bkietz** [comment]
@github-actions autotune everything

**Comment by kszucs** [comment]
We also need to re-enable the disabled ORC test case.

**Comment by kszucs** [comment]
The lint failure is unrelated and should be fixed on master.

**Comment by kszucs** [comment]
The ORC [test is passing](https://github.com/apache/arrow/pull/7048/checks?check_run_id=625878942#step:7:4498). Thanks @zhouyuan for fixing it!

**Review by kszucs — APPROVED** [review]

File: cpp/src/jni/orc/jni_wrapper.cpp [hunk 1]
```diff
@@ -276,9 +276,16 @@ Java_org_apache_arrow_adapter_orc_OrcStripeReaderJniWrapper_next(JNIEnv* env,
 
   for (size_t j = 0; j < buffers.size(); ++j) {
     auto buffer = buffers[j];
+    uint8_t* data = nullptr;
+    int size = 0;
```
Comment (fsaintjacques) [inline cpp/src/jni/orc/jni_wrapper.cpp:280]: That should be int64_t.

**Pull request closed without merging**